### PR TITLE
Fix line continuation parsing, follow tree-sitter-python and make it a visible token 

### PR DESCRIPTION
**Pull request by NathanLovato**
This fixes today's earlier implementation of the \ token. First, I can't use "\\" in tree sitter queries, at least with topiary for GDScript formatting: it causes the query to fail to compile.

Then I realized that with the approach I took, the line continuation rule would need to be added everywhere. Initially I thought of putting it in the scanner.c file, but ended up checking what Python did and it turns out that when you put something in extras you don't necessarily need to parse them with the external scanner which was a misconception I had.

Long story short, this copies what Python does. GDscript has similar rules to Python for trailing \ with some really rare edge cases like the get node case. This seems to work well. Now it'll also be a named node that'll be usable in the formatter, there again, same as the Python tree sitter parser.

### CODE REVIEW

File: test/corpus/linecontinuations.txt [hunk 1]
```diff
@@ -26,12 +28,121 @@ world"
...
 
 ---
 
 (source
+  (comment)
   (expression_statement
-  (get_node
-    (ERROR))))
+    (get_node
+      (line_continuation))))
```
Comment (NathanLovato) [inline test/corpus/linecontinuations.txt:41]: This will not parse in Godot for some weird reason (otherwise line continuation marks are very permissive and can be put nearly anywhere after a token), but in this version of the parser it is now a valid sequence. 

I went with updating the test for simplicity because this looks more like a bug in Godot than something that should error out.

Should it be an explicit error instead?
